feat: Add default currency for revenue tracking

rafaeelaudibert opened the pull request:
This can now be set both in the `Revenue` page and also on the Web Analytics settings.

This will be later used by us to convert all of the revenue information we're chugging in on the Web Analytics dashboard

Screenshots? See storybook

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Manually + storybook

rafaeelaudibert committed:
feat: Add default currency for revenue tracking

This can now be set both in the `Revenue` page and also on the Web
Analytics settings.

This will be later used by us to display revenue information on the Web
Analytics dashboard

rafaeelaudibert committed:
feat: Display revenue on the chosen currency

We're now displaying revenue on the currency provided by our customers. We're not converting to that currency yet, this is coming soon

rafaeelaudibert committed:
docs: Note how conversion works

robbie-c commented on frontend/src/scenes/data-management/revenue/CurrencyDropdown.tsx:
```diff
@@ -0,0 +1,351 @@
+    [SupportedCurrencies.WST]: '🇼🇸',
+    [SupportedCurrencies.XAF]: '🌍', // Central Africa currency, using Africa+Europe globe emoji
+    [SupportedCurrencies.XCD]: '🌎', // East Caribbean Dollar, using Americas globe emoji
+    [SupportedCurrencies.XDR]: '🌎', // Special Drawing rights, IMF "currency", Americas globe emoji because I have to pick something
```
If this comes up in prod, I will pay you one XDR

rafaeelaudibert replied:
I could just remove this one right......

rafaeelaudibert replied:
I removed it anyway 🤣

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

rafaeelaudibert commented on frontend/src/queries/schema/schema-general.ts:
```diff
@@ -2528,8 +2528,168 @@ export type CachedTracesQueryResponse = CachedQueryResponse<TracesQueryResponse>
+
+// NOTE: Keep in sync with posthog/models/exchange_rate/currencies.py
+// to provide proper type safety for the baseCurrency field
+export enum SupportedCurrencies {
```
@robbie-c You've had more experience with the codebase, should this be in the singular instead?

rafaeelaudibert committed:
refactor: Remove XDR

It doesn't make any sense, nobody's tracking this lol

rafaeelaudibert committed:
refactor: Simplify TS casts

rafaeelaudibert committed:
docs: Properly capitalize PostHog

rafaeelaudibert committed:
fix: Fix incorrect name for `TRY` currency

For some reason we can't have an enum with a constant called TRY, so our generation is generating `TRY_` instead, let's use it like that

rafaeelaudibert committed:
fix: Fix TS error

rafaeelaudibert committed:
tests: Regenerate snapshot

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

robbie-c reviewed (approved):
From a UI point of view, I think we could do more to communicate that if an event doesn't have a currency property, or the value is null / invalid, that we would assume that it's dollars.

<img width="955" alt="Screenshot 2025-03-06 at 13 23 46" src="https://github.com/user-attachments/assets/be7c40bc-f0d2-4c57-8cce-81635fcae51e" />

Also, it'd be cool if the table could should show the currency value, and also the revenue after being converted to the team's currency. Feel free to skip that for this PR though!

Other than that LGTM!

rafaeelaudibert commented:
@robbie-c 

> From a UI point of view, I think we could do more to communicate that if an event doesn't have a currency property, or the value is null / invalid, that we would assume that it's dollars.

Yeah, I have some stuff inside the small info tooltip but I might bring some of it outside. We actually aren't assuming it's dollars, we're assuming it's whatever the base currency is.

> Also, it'd be cool if the table could should show the currency value, and also the revenue after being converted to the team's currency. Feel free to skip that for this PR though!

I will do this once https://github.com/PostHog/posthog/pull/29557 is merged! We aren't really converting them yet.

lricoy reviewed (approved):
LGTM, Robbie beated me to it, just +1 on moving the enums/dicts/maps closer together to countryCodes.

rafaeelaudibert committed:
refactor: Use `CurrencyCode` rather than `SupportedCurrencies`

Just looks nicer

rafaeelaudibert committed:
refactor: Extract geography-related to code to new utils folder

This includes country and currency stuff
